platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/ApplicationProcessor.java

```diff
@@ -0,0 +1,220 @@
+    LOGGER.debug("Processing applications import");
+    final Map<String, JsonApplication> jappsMap =
+        japps
+            .stream()
+            .collect( // linked map to preserve order
+                org.codice.ddf.admin.application.service.migratable.Collectors.toLinkedMap(
+                    JsonApplication::getName, Function.identity()));
+
+    // populate the task list
+    processMemoryApplications(jappsMap, tasks);
+    return processLeftoverExportedApplications(report, jappsMap, tasks);
+  }
+
+  /**
+   * Processes applications in memory by recording tasks to start applications that were originally
+   * started.
+   *
+   * <p><i>Note:</i> Any applications found in memory are removed from the provided map since they
+   * have been processed.
+   *
+   * @param japps the set of apps on the original system
+   * @param tasks the task list where to record tasks to be executed
+   */
+  @SuppressWarnings("squid:S3776" /* additional branches used only for tracing */)
+  public void processMemoryApplications(Map<String, JsonApplication> japps, TaskList tasks) {
+    LOGGER.debug("Processing applications defined in memory");
+    // check all apps in memory and either start or stop them based on the exported info
+    for (final Application app : service.getApplications()) {
+      final String name = app.getName();
+      final JsonApplication japp = japps.remove(name); // remove from japps when we find it
+
+      processApplication(japp, app, tasks);
+    }
+  }
+
+  /**
+   * Processes applications that were left over after having dealt with what is in memory. The
+   * implementation will try to find the missing applications (although unlikely to be found). If
+   * found, they will be processed by recording tasks to start them if they were originally started.
+   *
+   * @param report the report where to record errors
+   * @param japps the set of apps on the original system that were not yet processed from memory
+   * @param tasks the task list where to record tasks to be executed
+   * @return <code>false</code> if we failed to retrieve from memory a left over application; <code>
+   *     true</code> if everything was ok
+   */
+  public boolean processLeftoverExportedApplications(
+      ProfileMigrationReport report, Map<String, JsonApplication> japps, TaskList tasks) {
+    LOGGER.debug("Processing leftover exported applications");
+    boolean processed = true; // until proven otherwise
+
+    // check if there are anything left from the exported info that should be started or that is
+    // not installed
+    for (final JsonApplication japp : japps.values()) {
+      final String name = japp.getName();
+
+      if (japp.isStarted()) {
+        final Application app = getApplication(report, name);
+
+        if (app == null) { // app not available!!!
+          // error already recorded from getApplication()
+          processed = false;
```

Comment:
I am ok to remove the comment since in the original version there were other branches that would record errors where this one wasn't so the comment was more useful back then. 
Fine to rename the variable.